schema/me/conversation.js

```diff
@@ -23,9 +143,13 @@ export default {
         client_application_id: IMPULSE_APPLICATION_ID,
       })
       .then(data => {
-        return impulse.with(data.token, { method: "GET" })(`conversations/${id}`).then(impulseData => {
-          return impulseData
-        })
+        return impulse
+          .with(data.token, { method: "GET" })(`conversations/${id}`, {
+            expand: ["messages"],
```

Comment:
We should indeed not request a lot of data to be sent over the wire if we’re not going to use it. I’m not sure I fully understand the context of the question, do you mean something like this?

```js
return impulse
  .with(data.token, { method: "GET" })(`conversations/${id}`, {
    expand: queryingForMessages ? ["messages"] : null,
  })
```

That is, assuming that this leads to a GET request that MP properly serialises and impulse properly handles. Otherwise:

```js
const params = {}
if (queryingForMessages) {
  params.expand = ["messages"]
}
return impulse
  .with(data.token, { method: "GET" })(`conversations/${id}`, params)
```